sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -448,7 +455,18 @@ export class ApplicationService extends RecordDataModelService<Application> {
       associatedFiles,
     );
 
-    return this.repo.save(draftApplication);
+    const savedDraftApplication = await this.repo.save(draftApplication);
+    // Update application version properties for new draft application.
+    if (isNewDraftApplication) {
+      await this.repo.update(
+        { id: savedDraftApplication.id },
+        {
+          precedingApplication: { id: savedDraftApplication.id } as Application,
+          parentApplication: { id: savedDraftApplication.id } as Application,
```

Comment:
I believe type casting `as Application` is not required here.

![image](https://github.com/user-attachments/assets/4b976fd1-73c1-45d7-ae73-2a0d0bcdec4f)